File: 2255-count-prefixes-of-a-given-string/2255-count-prefixes-of-a-given-string.py
class Solution:
    def isPrefix(self, word: str, s: str) -> bool:
        n = len(word)
        if n > len(s):
            return False
        for i in range(n):
            if word[i] != s[i]:
                return False
        return True
    def countPrefixes(self, words: List[str], s: str) -> int:
        count = 0
        for word in words:
            if self.isPrefix(word, s):
                count += 1
        return count

File: 2255-count-prefixes-of-a-given-string/2255-count-prefixes-of-a-given-string.cpp
class Solution {
public:
    bool isPrefix(string word, string s) {
        int n = word.length();
        if (n > s.length()) {
            return false;
        }
        for (int i=0; i<n; i++) {
            if (word[i] != s[i]) {
                return false;
            }
        }
        return true;
    }
public:
    int countPrefixes(vector<string>& words, string s) {
        int count = 0;
        int wordsSize = words.size();
        string word;
        for (int i=0; i<wordsSize; i++) {
            word = words[i];
            if (this->isPrefix(word, s)) {
                count++;
            }
        }
        return count;
    }
};